lib/ecto/changeset.ex

```diff
@@ -1275,6 +1276,62 @@ defmodule Ecto.Changeset do
 
   ## Validations
 
+    @doc """
+  Returns a keyword list of the validations for this changeset.
+
+  The keys in the list are the names of fields, and the values are a
+  validation associated with the field. A field may occur multiple
+  times in the list.
+
+  A quick example:
+
+      iex> %Post{}    |>
+      ...> change()   |>
+      ...> validate_format(:title, ~r/^\w+:\s/,
+      ...>      message: "must start with a topic")  |>
+      ...> validate_length(:title, max: 100)
+      ...> validation_specs()
+      [
+        title: { :length, [ max: 100 ]},
+        title: { :format, ~r/^\w+:\s/ }
+      ]
+
+  The following validations may be included in the result. 
+  
+  This first group contains validations that take a keyword list of validators,
+  where the validators are show immediately following the validation type. 
+  This list may also include a `message:` key.
+
+    * `{ :length, [ «validators»… ] }`
+
+      * `min: n`
+      * `max: n`
+      * `is: n`
+      * `count: :graphemes | :codepoints`
+
+  * `{ :number,  [ «validators»… ] }`
+    
+     * `equal_to: n`
+     * `greater_than: n`
+     * `greater_than_or_equal_to: n`
+     * `less_than: n`
+     * `less_than_or_equal_to: n`
+
+  The other validators simply take a value:
+
+  * `{ :exclusion, «Enum.t» }`
+  * `{ :format, ~r/pattern/ }`
+  * `{ :inclusion, «Enum.t» }`
+  * `{ :subset, «Enum.t» }`
```

Comment:
I'm not sure about documenting all the possible values this way, since it's generally open-ended. The values provided here are generally supplied by the `validate_change/4` function. If you have a custom validation function (or things from some library) it can put just about any value in there - documenting only the things provided by the `Ecto.Changeset` module itself might be surprising for the user when they get something different.

Reply:
I think it is good to document them but we should make it clear that any value can be added and that it is open ended.